spec/unit/resolver_spec.rb

```diff
@@ -377,6 +377,33 @@ def create_resolver(podfile = @podfile, locked_deps = empty_graph, specs_updated
         EOS
       end
 
+      # TODO: Make this work
+      it 'raises with extra metadata if it finds two conflicting explicit dependencies which are :podspec' do
+        podspec_path = fixture('integration/Reachability/')
+        podfile = Podfile.new do
+          platform :ios
+          pod 'Reachability', :podspec => podspec_path
+        end
+
+        locked_dep = Dependency.new('Reachability')
+        locked_dep.external_source = { :podspec => podspec_path }
+        locked_deps = dependency_graph_from_array([locked_dep])
+
+        spec = Spec.new do |s|
+          s.name         = 'Reachability'
+          s.version      = '2.9.0' # it's 3.0.0 in the integration fixture
+        end
+        config.sandbox.expects(:specification).with('Reachability').returns(spec)
+
+        resolver = create_resolver(podfile, locked_deps)
+        e = lambda { resolver.resolve }.should.raise Informative
+
+        e.message.should.include <<-EOS.strip
+It seems like you've changed the version of the dependency `Reachability`
+and it differs from the version stored in `Pods/Local Podspecs`.\nYou should run `pod update Reachability --no-repo-update` to apply
+        EOS
+      end
```

Comment:
I'm a little surprised I can't get this test working, what I think it does:

- Sets up Reachability as the fixtured version (3.0.0)
- Sets up a lockfile for Reachability which references the same on-disk Podspec (3.0.0)
- Tells the sandbox to actually return a version of (2.9.0)

Which I'm sure is the case for us locally, is there anything obvious I'm missing (the resolve doesn't raise in this case)

Reply:
We took care of it.